### PR TITLE
UCT/CMA/STRING: Print IO vector in case of error

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucs/datastruct/string_buffer.c [hunk 1]
```diff
@@ -125,6 +125,18 @@ void ucs_string_buffer_append_flags(ucs_string_buffer_t *strb, uint64_t mask,
...
+void ucs_string_buffer_append_iovec(ucs_string_buffer_t *strb,
+                                    const struct iovec *iov, size_t iovcnt)
+{
+    size_t iov_index;
+
+    for (iov_index = 0; iov_index < iovcnt; ++iov_index) {
+        ucs_string_buffer_appendf(strb, "%p,%zu|", iov[iov_index].iov_base,
+                                  iov[iov_index].iov_len);
+    }
+    ucs_string_buffer_rtrim(strb, "|");
```
Comment (hoopoepg) [inline src/ucs/datastruct/string_buffer.c:137]: in case if you call this function twice you get unreadable string like
```0x12345678,1230x56781234,456```

Comment (yosefe) [reply src/ucs/datastruct/string_buffer.c:137]: yes, the user is responsible to insert separators in between if needed

File: src/uct/sm/scopy/cma/cma_ep.c [hunk 2]
```diff
@@ -56,21 +58,45 @@ UCS_CLASS_DEFINE(uct_cma_ep_t, uct_scopy_ep_t)
...
+    ucs_status_t status;
+
+    status  = uct_iface_handle_ep_err(&iface->super, &ep->super.super.super,
+                                      UCS_ERR_CONNECTION_RESET);
+    log_lvl = uct_base_iface_failure_log_level(iface, status,
+                                               UCS_ERR_CONNECTION_RESET);
+
+    /* Dump IO vector */
+    ucs_string_buffer_append_iovec(&local_iov_str, local_iov, local_iov_cnt);
+    ucs_string_buffer_append_iovec(&remote_iov_str, remote_iov, 1);
```
Comment (hoopoepg) [inline src/uct/sm/scopy/cma/cma_ep.c:81]: > yes, the user is responsible to insert separators in between if needed

ok :)

Comment (hoopoepg) [reply src/uct/sm/scopy/cma/cma_ep.c:81]: may be add separator here?

**Review by brminich — APPROVED** [review]

**Review by dmitrygx — APPROVED** [review]

**Pull request merged**